### PR TITLE
Bgshacklett's "Add events to the site" PR

### LINKED ISSUE #3: calendar feature
# calendar feature

## User Story

Would like calendar display for users to see new events

## Tasks

- [ ] setup calendar feature

## Extra Notes

* https://levelup.gitconnected.com/building-a-calendar-with-eleventy-and-css-grid-febef21eb246
* depends on #2

### PR DESCRIPTION
**Pull request by mdaverde**
From  @bgshacklett's PR: 

> Once finished, this PR will add:
> 
> * an events overview page
> * a page for events by "host" (Orlando Developers, Orlando DevOps, Orlando JS, etc.).
> * automatically generated pages per event
> 
> Known open issues:
> 
> * [x]  Update styling for mobile devices.
> * [ ]  Test on Chrome and Safari (Firefox FTW!).
> * [x]  Redesign the single-event page.
> * [ ]  The heading of the event card sits too low compared to the date stamp.

Closes #3

### CODE REVIEW

File: src/_data/events.yaml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        - "#orlandodevs"
+
+    - title: Orlando Devs Holiday Party
+      imageUrl: "https://img.evbuc.com/https%3A%2F%2Fcdn.evbuc.com%2Fimages%2F561994959%2F930410036073%2F1%2Foriginal.20230725-195128?w=940&auto=format%2Ccompress&q=75&sharp=10&rect=0%2C0%2C1200%2C600&s=566c8ea2fde84eed55214002ded53c66"
+      datetime: "2024-02-14T1906-0500"
+      duration: 1.5h
+      priceInDollars: 0
+
+
+- host: bar
```
Comment (wgeorgecook) [inline src/_data/events.yaml:83]: I think the existing events is a good enough template. We can either take these out or provide an `events_template.yaml` file for examples.

**Comment by lastlink** [comment]
related to #3

File: .eleventy.js [hunk 2]
```diff
@@ -1,16 +1,75 @@
 const yaml = require("js-yaml");
 const { DateTime } = require("luxon");
+
 const syntaxHighlight = require("@11ty/eleventy-plugin-syntaxhighlight");
+const pluginWebc      = require("@11ty/eleventy-plugin-webc");
+
 const htmlmin = require("html-minifier");
 
+const filters = require('./src/js/filters.js');
+
```
Comment (wgeorgecook) [inline .eleventy.js:10]: Maybe we should've added a linter to the scope 😬 thems a lot of newlines.

File: src/_data/flat_events.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    const fileContents = fs.readFileSync(filePath, 'utf8');
+    const data = yaml.load(fileContents);
+    let flatEvents = [];
+
+    data.forEach(hostItem => {
+      hostItem.events.forEach(event => {
+        // Optionally, add host information to each event
+        event.host = hostItem.host;
+        event.bannerImg = hostItem.banner_img;
+        console.log(event);
```
Comment (wgeorgecook) [inline src/_data/flat_events.js:17]: If console messages are helpful I would prefer debug to log so they don't clog up the console by default.

File: src/_includes/partials/event_card.html [hunk 4]
```diff
@@ -0,0 +1,161 @@
+<style>
```
Comment (wgeorgecook) [inline src/_includes/partials/event_card.html:1]: I think the style would be better in `static/css/`

**Review by snohio — APPROVED** [review]
:shipit: GTG

File: src/_data/events.yaml (hunk 1 above)
Comment (wgeorgecook) [reply src/_data/events.yaml:83]: Alternatively as a docstring or something at the top.

File: src/static/img/default_event_banner.png
Comment (wgeorgecook) [inline src/static/img/default_event_banner.png:1]: Perfect 👨🏻‍🍳 💋

**Review by wgeorgecook — APPROVED** [review]
lgtm 👍🏻

**Pull request merged**